enginetest/join_op_tests.go

```diff
@@ -2046,6 +2046,26 @@ WHERE
 			},
 		},
 	},
+	{
+		name: "join with % condition",
+		setup: [][]string{
+			{
+				"create table t1(c0 int)",
+				"create table t2(c0 int)",
+				"insert into t1 values (1),(2)",
+				"insert into t2 values (3),(4)",
+			},
+		},
+		tests: []JoinOpTests{
+			{
+				Query: `select * from t1 join t2 on (t1.c0 % 2) = (t2.c0 % 2)`,
+				Expected: []sql.Row{
+					{1, 3},
+					{2, 4},
+				},
+			},
+		},
+	},
```

Comment:
Write some more tests where the columns are of Decimal Type with varying scale and precision.
Want to make sure trailing 0s and rounding/truncating behavior matches MySQL.